deploy-cloudrun/src/cloudRun.ts

```diff
@@ -0,0 +1,258 @@
+    const listRequest: run_v1.Params$Resource$Projects$Locations$Services$List = {
+      parent: this.parent,
+      auth: authClient,
+    };
+    const serviceListResponse: GaxiosResponse<run_v1.Schema$ListServicesResponse> = await this.run.projects.locations.services.list(
+      listRequest,
+      this.methodOptions,
+    );
+    const serviceList: run_v1.Schema$ListServicesResponse =
+      serviceListResponse.data;
+    let serviceNames: string[] = [];
+    if (serviceList.items !== undefined) {
+      serviceNames = serviceList.items!.map(
+        (service: run_v1.Schema$Service) => service.metadata!.name as string,
+      );
+    }
+    return serviceNames;
+  }
+
+  /**
+   * Deploy a Cloud Run service.
+   *
+   * @param service Service object
+   * @returns Service object
+   */
+  async deploy(service: Service): Promise<run_v1.Schema$Service> {
+    const authClient = await this.getAuthClient();
+    const serviceNames = await this.listServices();
+    let serviceResponse: GaxiosResponse<run_v1.Schema$Service>;
+    if (serviceNames!.includes(service.name)) {
+      core.info('Creating a service revision...');
+      // Replace service
+      const createServiceRequest: run_v1.Params$Resource$Projects$Locations$Services$Replaceservice = {
+        name: this.getResource(service),
+        auth: authClient,
+        requestBody: service.request,
+      };
+      serviceResponse = await this.run.projects.locations.services.replaceService(
+        createServiceRequest,
+        this.methodOptions,
+      );
+    } else {
+      core.info('Creating a new service...');
+      // Create service
+      const createServiceRequest: run_v1.Params$Resource$Projects$Locations$Services$Create = {
+        parent: this.parent,
+        auth: authClient,
+        requestBody: service.request,
+      };
+      serviceResponse = await this.run.projects.locations.services.create(
+        createServiceRequest,
+        this.methodOptions,
+      );
+    }
+    core.info(`Service ${service.name} has been successfully deployed.`);
+    return serviceResponse.data;
+  }
+
+  /**
+   * Deletes a Cloud Run service.
+   *
+   * @param service Service object
+   */
+  async delete(service: Service): Promise<void> {
+    const authClient = await this.getAuthClient();
+    try {
+      const deleteServiceRequest: run_v1.Params$Resource$Projects$Locations$Services$Delete = {
+        name: this.getResource(service),
+        auth: authClient,
+      };
+      await this.run.projects.locations.services.delete(
+        deleteServiceRequest,
+        this.methodOptions,
+      );
+      core.info(`Service ${service.name} has been successfully deleted.`);
+    } catch (e) {
+      core.info(`Error deleting Service ${service.name}: ` + e);
+    }
+  }
+
+  /**
+   * Set's IAM policy for service (Not Recommended)
+   *
+   * @param service Service object
+   */
+  async allowUnauthenticatedRequests(service: Service): Promise<void> {
```

Comment:
Are we supporting this?

Reply:
Not supporting, but I was keeping for prosperity. Should I remove?